--- a/game/day1.rpy
+++ b/game/day1.rpy
@@ -49,7 +49,7 @@ label intro:
 
     """
     play music distant_rumble
-    scene cg adelle with Dissolve(3):
+    scene cg adelle with Dissolve(3)
     show bubble
     show bubble2
     $renpy.pause(1)
@@ -534,7 +534,7 @@ label miriam_initial:
     "Her figure stood in that unfamiliar lab, her jagged image some twisted mockery of what I’m supposed to be. {p} But the memory doesn’t linger too long before light threatens to invade my sealed eyelids."
 
     window hide
-    scene bg miriamroom with Dissolve(3):
+    scene bg miriamroom with Dissolve(3)
 
     "I’m flat on my back on a plush bed. {w}I glance to my right, honing the source of the noise."
 
@@ -2645,7 +2645,7 @@ label OH_SHIT:
 
 label akira_initial:
 
-    scene bg miriamfirstfloor night:
+    scene bg miriamfirstfloor night
     show screen clock
     stop music
     play sound doki
@@ -2831,7 +2831,7 @@ label akira_initial:
 
     """
     hide black with dissolve
-    show bg park night 2 behind distortion with ImageDissolve("bg park night 2.png",2):
+    show bg park night 2 behind distortion with ImageDissolve("bg park night 2.png",2)
 
     "It’s when my sight returns again in some empty park that finally I'm hit by an indisputable truth--that {i}all of it{/i}{w=0.2} is real.{w} The duel I woke up from, my untimely demise, this shabby town, my exhausting existence, these senses that invade me and overflow, I know I am Adelle."
     play music submarine_magma
@@ -3094,7 +3094,7 @@ label akirainitial_2:
 
     stop music fadeout 2.0
     play nature nightambience
-    scene bg park night 2 with fadehold:
+    scene bg park night 2 with fadehold
     $renpy.pause(2)
     #show blue:
     #    alpha 0.3
--- a/game/day2.rpy
+++ b/game/day2.rpy
@@ -1,7 +1,7 @@
 label myoelectric_morning:
     $ day = 2
     $ mm = "[[ Miriam ]"
-    scene black with blinds:
+    scene black with blinds
     scene bg sky day with dissolve
     show screen clock with e
     play ambience "audio/se/morning_streets.mp3"
@@ -3090,7 +3090,7 @@ label tactical_hallucination:
     play sound blow3
     d"Son of a bitch, I'm not the chick-magnet protagonist at all. . . I'm the skirt flip girl?"
     play sound button06
-    show akjk 3 with e:
+    show akjk 3 with e
     a"Adelle, this fantasy is way too vague. You're never going to remember me by dreaming up fake galge in your head. At this rate, you'll probably end up choosing a different route, anyway."
     show adjk -3 with e
     d"Says who? You're the one who led this on."
--- a/game/day6.rpy
+++ b/game/day6.rpy
@@ -4068,7 +4068,7 @@ label forlove:
     l"“Hm . . . Well, it was you who gave me pointers on my skincare.”"
     m"“Ah, so you felt it, after all.”"
     l"“No, those ugly gloves gave it away the moment you walked through the door.”"
-    show cg 2 with dissolve:
+    show cg 2 with dissolve
     l"“I’m leaving. I don’t have any stake in this war anymore, and I expect I won’t ever hear about it again. But if you intend to end it like {i}that,{/i} I'll warn you: just quit while you're ahead.”"
     l"“You're young and have dreams, I understand. You're a lucky and smart girl for making it this far, but there's always limits to what we humans can achieve.”"
     l"“Using your {i}own body{/i} to test the Arbiter antibodies. Nah, just another self-righteous plight of yours. You’re no hero, you're not noble. And I don't want your filthy DNA near any of my girls. ”"
@@ -4289,7 +4289,7 @@ label for_eternity:
         repeat
     s"“Damn it all!! Who the hell designed this terrible security system -- !!”"
     pause 1
-    show cg 2slqu 4 behind movie with fade:
+    show cg 2slqu 4 behind movie with fade
     s"“No, I can't be mad at them . . Not them. . .But you, idiot. . .”"
     show cg 2slqu 3 at truecenter with fade:
         zoom 0.35
