From bec90d72d2ce37a8d78175a14e9f47fe9e8790f4 Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Sun, 17 Mar 2024 13:24:35 -0400
Subject: [PATCH] Allow parellel install of wlroots versions

--- a/include/meson.build
+++ b/include/meson.build
@@ -25,7 +25,7 @@ if not features.get('session')
 endif
 
 install_subdir('wlr',
-	install_dir: get_option('includedir'),
+	install_dir: '@0@/@1@'.format(get_option('includedir'), versioned_name),
 	exclude_files: exclude_files,
 )
 
--- a/include/wlr/meson.build
+++ b/include/wlr/meson.build
@@ -22,4 +22,4 @@ ver_h = configure_file(
 	configuration: version_data,
 )
 
-install_headers(conf_h, ver_h, subdir: 'wlr')
+install_headers(conf_h, ver_h, subdir: '@0@/wlr'.format(versioned_name))
--- a/meson.build
+++ b/meson.build
@@ -16,6 +16,7 @@ version_major = version.split('.')[0]
 version_minor = version.split('.')[1]
 assert(version_major == '0')
 soversion = version_minor.to_int() - 5
+versioned_name = '@0@-@1@.@2@'.format(meson.project_name(), version_major, version_minor)
 
 little_endian = target_machine.endian() == 'little'
 big_endian = target_machine.endian() == 'big'
@@ -174,7 +175,7 @@ wlr_inc = include_directories('include')
 symbols_file = 'wlroots.syms'
 symbols_flag = '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), symbols_file)
 lib_wlr = library(
-	meson.project_name(), wlr_files,
+	versioned_name, wlr_files,
 	soversion: soversion.to_string(),
 	dependencies: wlr_deps,
 	include_directories: [wlr_inc],
@@ -207,7 +208,9 @@ endif
 pkgconfig = import('pkgconfig')
 pkgconfig.generate(
 	lib_wlr,
+	name: versioned_name,
 	description: 'Wayland compositor library',
+	subdirs: versioned_name,
 	url: 'https://gitlab.freedesktop.org/wlroots/wlroots',
 	variables: wlr_vars,
 )
-- 
2.44.0

