From 2a266822cbaee0dbc820f034a959ec84dbd1d364 Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Wed, 9 Aug 2023 19:59:23 -0400
Subject: [PATCH] Port check for --default-symver to autoconf

Signed-off-by: Violet Purcell <vimproved@inventati.org>
--- a/configure.ac
+++ b/configure.ac
@@ -25,6 +25,9 @@ AC_TYPE_UINT16_T
 AC_TYPE_UINT8_T
 AC_TYPE_LONG_LONG_INT
 
+jansson_soversion="4"
+AC_SUBST([jansson_soversion])
+
 AC_C_INLINE
 case $ac_cv_c_inline in
     yes) json_inline=inline;;
@@ -139,7 +142,10 @@ AC_SUBST(JSON_BSYMBOLIC_LDFLAGS)
 
 # Enable symbol versioning on GNU libc
 JSON_SYMVER_LDFLAGS=
-AC_CHECK_DECL([__GLIBC__], [JSON_SYMVER_LDFLAGS=-Wl,--default-symver])
+AC_CHECK_DECL([__GLIBC__],
+  [AX_CHECK_LINK_FLAG([-Wl,--default-symver],
+    [JSON_SYMVER_LDFLAGS=-Wl,--default-symver],
+    [JSON_SYMVER_LDFLAGS=-Wl,--version-script,$ac_abs_confdir/jansson.sym])])
 AC_SUBST([JSON_SYMVER_LDFLAGS])
 
 AC_ARG_ENABLE([ossfuzzers],
@@ -168,6 +174,7 @@ AC_SUBST([AM_CFLAGS])
 
 AC_CONFIG_FILES([
         jansson.pc
+	jansson.sym
         Makefile
         doc/Makefile
         src/Makefile
--- /dev/null
+++ b/jansson.sym.in
@@ -0,0 +1,5 @@
+JANSSON_@jansson_soversion@ {
+	global:
+	*;
+};
+
-- 
2.41.0

