From 0a8c3c8e7e568656da8dfb5dd8e882a80f363099 Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Thu, 27 Jul 2023 21:36:59 -0400
Subject: [PATCH] un-force bfd and libgcc

--- a/src/boot/efi/meson.build
+++ b/src/boot/efi/meson.build
@@ -151,11 +151,6 @@ if get_option('mode') == 'developer' and get_option('debug')
 endif
 
 efi_c_ld_args = [
-        # We only support bfd. gold is going away, lld has issues with LTO on x86
-        # and mold does not support linker scripts.
-        '-fuse-ld=bfd',
-
-        '-lgcc',
         '-nostdlib',
         '-static-pie',
         '-Wl,--entry=efi_main',
-- 
2.41.0

