From 61af57bd8c763ea088bdfb75a24768e70414624b Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Thu, 27 Jul 2023 22:19:35 -0400
Subject: [PATCH] link kernel-install statically

--- a/meson.build
+++ b/meson.build
@@ -4419,7 +4419,7 @@ kernel_install = executable(
         'kernel-install',
         'src/kernel-install/kernel-install.c',
         include_directories : includes,
-        link_with : [libshared],
+        link_with : [libsystemd_static, libshared_static],
         dependencies : [userspace,
                         versiondep],
         install_rpath : rootpkglibdir,
-- 
2.41.0

