From 8b50dd33cc35a12a13f491b75731031c50a5cf5e Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Fri, 30 Jun 2023 23:07:19 +0000
Subject: [PATCH] Unbundle dependencies

---
 Makefile     | 12 ++++++------
 src/Makefile |  2 +-
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index fa951df..c1e9dba 100644
--- a/Makefile
+++ b/Makefile
@@ -7,8 +7,8 @@ CC ?= cc
 CFLAGS += -I$(DEP)/include -I$(SRC)/include -I$(SRC) -D_XOPEN_SOURCE=500
 CFLAGS += -Wall -Wextra -Wno-unused-parameter -DPALERAIN_VERSION=\"2.0.0\"
 CFLAGS += -Wno-unused-variable -I$(SRC)/src -std=c99 -D_C99_SOURCE -D_POSIX_C_SOURCE=200112L
-LIBS += $(DEP)/lib/libimobiledevice-1.0.a $(DEP)/lib/libirecovery-1.0.a $(DEP)/lib/libusbmuxd-2.0.a
-LIBS += $(DEP)/lib/libimobiledevice-glue-1.0.a $(DEP)/lib/libplist-2.0.a -pthread -lm
+LIBS += -limobiledevice-1.0 -lirecovery-1.0 -lusbmuxd-2.0
+LIBS += -limobiledevice-glue-1.0 -lplist-2.0 -lpthread -lm
 ifeq ($(TARGET_OS),)
 TARGET_OS = $(shell uname -s)
 endif
@@ -21,9 +21,9 @@ LDFLAGS += -Wl,-dead_strip
 LIBS += -framework CoreFoundation -framework IOKit
 else
 CFLAGS += -fdata-sections -ffunction-sections
-LDFLAGS += -static -no-pie -Wl,--gc-sections
+LDFLAGS += -no-pie -Wl,--gc-sections
 endif
-LIBS += $(DEP)/lib/libmbedtls.a $(DEP)/lib/libmbedcrypto.a $(DEP)/lib/libmbedx509.a $(DEP)/lib/libreadline.a
+LIBS += -lmbedtls -lmbedcrypto -lmbedx509 -lreadline
 
 ifeq ($(TUI),1)
 LIBS += $(DEP)/lib/libnewt.a $(DEP)/lib/libpopt.a $(DEP)/lib/libslang.a
@@ -68,7 +68,7 @@ export SRC DEP CC CFLAGS LDFLAGS LIBS TARGET_OS DEV_BUILD BUILD_DATE BUILD_TAG B
 
 all: palera1n
 
-palera1n: download-deps
+palera1n:
 	$(MAKE) -C src
 
 clean:
@@ -76,7 +76,7 @@ clean:
 	$(MAKE) -C docs clean
 
 download-deps:
-	$(MAKE) -C src checkra1n-macos checkra1n-linux-arm64 checkra1n-linux-armel checkra1n-linux-x86 checkra1n-linux-x86_64 checkra1n-kpf-pongo ramdisk.dmg binpack.dmg Pongo.bin
+	$(MAKE) -C src
 
 docs:
 	$(MAKE) -C docs
diff --git a/src/Makefile b/src/Makefile
index 2605c95..730a499 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -77,7 +77,7 @@ endif
 ifeq ($(USE_LIBUSB),1)
 OBJECTS += usb_libusb.o
 CFLAGS += -DUSE_LIBUSB=1
-LIBS += $(DEP)/lib/libusb-1.0.a
+LIBS += -lusb-1.0
 endif
 
 ifneq ($(ROOTFUL),0)
-- 
2.41.0

