--- a/src/update-mime-database.cpp
+++ b/src/update-mime-database.cpp
@@ -2158,7 +2158,7 @@ static void check_in_path_xdg_data(const char *mime_path)
 
 	env = getenv("XDG_DATA_DIRS");
 	if (!env)
-		env = "/usr/local/share/"PATH_SEPARATOR"/usr/share/";
+		env = "/usr/local/share/" PATH_SEPARATOR "/usr/share/";
 	dirs = g_strsplit(env, PATH_SEPARATOR, 0);
 	g_return_if_fail(dirs != NULL);
 	for (n = 0; dirs[n]; n++)
@@ -2170,7 +2170,7 @@ static void check_in_path_xdg_data(const char *mime_path)
 		dirs[n] = g_build_filename(g_get_home_dir(), ".local",
 						"share", NULL);
 	n++;
-	
+
 	for (i = 0; i < n; i++)
 	{
 		if (stat(dirs[i], &dir_info) == 0 &&
