From https://github.com/lsp-plugins/lsp-dsp-lib/commit/d9d18aa6113686e950133feab03fe956cd890fc6 Mon Sep 17 00:00:00 2001
From: sadko4u <sadko4u@gmail.com>
Date: Sat, 26 Aug 2023 14:03:10 +0300
Subject: [PATCH] Fix C compatibility

--- a/include/lsp-plug.in/dsp/common/bitmap/b1b8.h
+++ b/include/lsp-plug.in/dsp/common/bitmap/b1b8.h
@@ -28,7 +28,7 @@
  * Put the source 1-bit bitmap to destination 8-bit bitmap,
  * the result pixel value will be computed as: result[i] = src[i]
  */
-LSP_DSP_LIB_SYMBOL(void, bitmap_put_b1b8, LSP_DSP_LIB_TYPE(LSP_DSP_LIB_TYPE(bitmap_t)) *dst, const LSP_DSP_LIB_TYPE(bitmap_t) *src, ssize_t x, ssize_t y);
+LSP_DSP_LIB_SYMBOL(void, bitmap_put_b1b8, LSP_DSP_LIB_TYPE(bitmap_t) *dst, const LSP_DSP_LIB_TYPE(bitmap_t) *src, ssize_t x, ssize_t y);
 
 /**
  * Put the source 1-bit bitmap to destination 8-bit bitmap,
