diff -upr a/data/plugins/generated/compressor.lv2/compressor.cpp b/data/plugins/generated/compressor.lv2/compressor.cpp
--- a/data/plugins/generated/compressor.lv2/compressor.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/compressor.lv2/compressor.cpp	2023-05-01 15:26:46.290283864 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/cubic_distortion.lv2/cubic_distortion.cpp b/data/plugins/generated/cubic_distortion.lv2/cubic_distortion.cpp
--- a/data/plugins/generated/cubic_distortion.lv2/cubic_distortion.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/cubic_distortion.lv2/cubic_distortion.cpp	2023-05-01 15:27:07.943307553 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 
 #ifndef FAUSTCLASS 
 #define FAUSTCLASS cubic_distortion
diff -upr a/data/plugins/generated/flanger.lv2/flanger.cpp b/data/plugins/generated/flanger.lv2/flanger.cpp
--- a/data/plugins/generated/flanger.lv2/flanger.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/flanger.lv2/flanger.cpp	2023-05-01 15:27:27.663023762 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/gate_stereo.lv2/gate_stereo.cpp b/data/plugins/generated/gate_stereo.lv2/gate_stereo.cpp
--- a/data/plugins/generated/gate_stereo.lv2/gate_stereo.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/gate_stereo.lv2/gate_stereo.cpp	2023-05-01 15:27:52.479331051 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/highpass_filter.lv2/highpass_filter.cpp b/data/plugins/generated/highpass_filter.lv2/highpass_filter.cpp
--- a/data/plugins/generated/highpass_filter.lv2/highpass_filter.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/highpass_filter.lv2/highpass_filter.cpp	2023-05-01 15:27:41.079496432 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/lowpass_filter.lv2/lowpass_filter.cpp b/data/plugins/generated/lowpass_filter.lv2/lowpass_filter.cpp
--- a/data/plugins/generated/lowpass_filter.lv2/lowpass_filter.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/lowpass_filter.lv2/lowpass_filter.cpp	2023-05-01 15:28:06.492460438 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/parametric_eq.lv2/parametric_eq.cpp b/data/plugins/generated/parametric_eq.lv2/parametric_eq.cpp
--- a/data/plugins/generated/parametric_eq.lv2/parametric_eq.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/parametric_eq.lv2/parametric_eq.cpp	2023-05-01 15:30:04.490657808 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/peak_limiter.lv2/peak_limiter.cpp b/data/plugins/generated/peak_limiter.lv2/peak_limiter.cpp
--- a/data/plugins/generated/peak_limiter.lv2/peak_limiter.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/peak_limiter.lv2/peak_limiter.cpp	2023-05-01 15:30:15.417060797 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/phaser.lv2/phaser.cpp b/data/plugins/generated/phaser.lv2/phaser.cpp
--- a/data/plugins/generated/phaser.lv2/phaser.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/phaser.lv2/phaser.cpp	2023-05-01 15:30:26.236801488 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/smooth_delay.lv2/smooth_delay.cpp b/data/plugins/generated/smooth_delay.lv2/smooth_delay.cpp
--- a/data/plugins/generated/smooth_delay.lv2/smooth_delay.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/smooth_delay.lv2/smooth_delay.cpp	2023-05-01 15:30:35.266586411 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/triple_synth.lv2/triple_synth.cpp b/data/plugins/generated/triple_synth.lv2/triple_synth.cpp
--- a/data/plugins/generated/triple_synth.lv2/triple_synth.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/triple_synth.lv2/triple_synth.cpp	2023-05-01 15:30:45.666340189 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/wah4.lv2/wah4.cpp b/data/plugins/generated/wah4.lv2/wah4.cpp
--- a/data/plugins/generated/wah4.lv2/wah4.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/wah4.lv2/wah4.cpp	2023-05-01 15:30:58.386041134 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/white_noise.lv2/white_noise.cpp b/data/plugins/generated/white_noise.lv2/white_noise.cpp
--- a/data/plugins/generated/white_noise.lv2/white_noise.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/white_noise.lv2/white_noise.cpp	2023-05-01 15:28:18.712282027 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
diff -upr a/data/plugins/generated/zita_rev1.lv2/zita_rev1.cpp b/data/plugins/generated/zita_rev1.lv2/zita_rev1.cpp
--- a/data/plugins/generated/zita_rev1.lv2/zita_rev1.cpp	2023-04-28 22:07:12.000000000 -0400
+++ b/data/plugins/generated/zita_rev1.lv2/zita_rev1.cpp	2023-05-01 15:31:11.322405975 -0400
@@ -40,6 +40,7 @@ Compilation options: -a /usr/share/faust
    have at least the Boost headers installed somewhere on your include path
    (the Boost libraries aren't needed). */
 
+#include <cstdint>
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
@@ -734,7 +735,6 @@ void LV2UI::run() {}
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
 #include <math.h>
 
 #ifndef FAUSTCLASS 
