From fe3c0baacb39092e2a04d1d98968a79a1eecbb08 Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Tue, 19 Mar 2024 23:09:18 -0400
Subject: [PATCH] packagemanager.d: fix getPackagePath assert

---
 source/dub/packagemanager.d | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/dub/packagemanager.d b/source/dub/packagemanager.d
index eb17a9f2..59edbfc7 100644
--- a/source/dub/packagemanager.d
+++ b/source/dub/packagemanager.d
@@ -503,7 +503,7 @@ class PackageManager {
 	 */
 	package(dub) NativePath getPackagePath(PlacementLocation base, in PackageName name, string vers)
 	{
-		assert(this.m_repositories.length == 3, "getPackagePath called in bare mode");
+		assert(this.m_repositories.length >= 3, "getPackagePath called in bare mode");
 		return this.m_repositories[base].getPackagePath(name, vers);
 	}
 
-- 
2.44.0

