Only in b: .mypy_cache
diff -upr a/renpy.py b/renpy.py
--- a/renpy.py	2025-03-17 10:12:07.971551700 -0400
+++ b/renpy.py	2025-04-29 15:05:16.603836696 -0400
@@ -28,6 +28,7 @@
 from __future__ import print_function, absolute_import
 
 import os
+import site
 import sys
 import warnings
 
@@ -44,36 +45,10 @@ def path_to_gamedir(basedir, name):
         The basename of the executable, with the extension removed.
     """
 
-    # A list of candidate game directory names.
-    candidates = [ name ]
-
-    # Add candidate names that are based on the name of the executable,
-    # split at spaces and underscores.
-    game_name = name
-
-    while game_name:
-        prefix = game_name[0]
-        game_name = game_name[1:]
-
-        if prefix == ' ' or prefix == '_':
-            candidates.append(game_name)
-
-    # Add default candidates.
-    candidates.extend([ 'game', 'data', 'launcher/game' ])
-
-    # Take the first candidate that exists.
-    for i in candidates:
-
-        if i == "renpy":
-            continue
-
-        gamedir = os.path.join(basedir, i)
-
-        if os.path.isdir(gamedir):
-            break
-
-    else:
-        gamedir = basedir
+    try:
+        gamedir = sys.argv[1]
+    except IndexError:
+        gamedir = site.getsitepackages()[0] + "/launcher/game"
 
     return gamedir
 
@@ -86,7 +61,11 @@ def path_to_common(renpy_base):
         The absolute path to the Ren'Py base directory, the directory
         containing this file.
     """
-    path = renpy_base + "/renpy/common"
+    cwd = os.getcwd()
+    if os.path.isdir(cwd + "/renpy/common"):
+        path = cwd + "/renpy/common"
+    else:
+        path = site.getsitepackages()[0] + "/renpy/common"
 
     if os.path.isdir(path):
         return path
@@ -211,8 +190,7 @@ def path_to_renpy_base():
     Returns the absolute path to the Ren'Py base directory.
     """
 
-    renpy_base = os.path.dirname(os.path.abspath(__file__))
-    renpy_base = os.path.abspath(renpy_base)
+    renpy_base = site.getsitepackages()[0]
 
     return renpy_base
 
