From d6f808e38088d706d757932249cc21bbdef65c4f Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Fri, 20 Sep 2024 13:26:48 -0400
Subject: [PATCH] fix build with cython>=3

--- a/renpy/audio/renpysound.pyx
+++ b/renpy/audio/renpysound.pyx
@@ -95,7 +95,7 @@ cdef extern from "renpysound_core.h":
 
 
 from renpy.audio.filter cimport get_apply_audio_filter
-RPS_apply_audio_filter = <void (*)(object, float *, int, int, int)> get_apply_audio_filter()
+RPS_apply_audio_filter = <void (*)(object, float *, int, int, int) noexcept> get_apply_audio_filter()
 
 
 def check_error():
@@ -514,7 +514,7 @@ def set_generate_audio_c_function(fn):
         import ctypes
         fn = ctypes.cast(fn, ctypes.c_void_p).value
 
-    RPS_generate_audio_c_function = <void (*)(float *, int)> <uintptr_t> fn
+    RPS_generate_audio_c_function = <void (*)(float *, int) noexcept> <uintptr_t> fn
 
 
 # Store the sample surfaces so they stay alive.
--- a/renpy/text/ftfont.pyx
+++ b/renpy/text/ftfont.pyx
@@ -120,7 +120,7 @@ cdef bint is_zerowidth(unsigned int char):
 
     return False
 
-cdef unsigned long io_func(FT_Stream stream, unsigned long offset, unsigned char *buffer, unsigned long count):
+cdef unsigned long io_func(FT_Stream stream, unsigned long offset, unsigned char *buffer, unsigned long count) noexcept:
     """
     Seeks to offset, and then reads count bytes from the stream into buffer.
     """
@@ -157,7 +157,7 @@ cdef unsigned long io_func(FT_Stream stream, unsigned long offset, unsigned char
 
     return count
 
-cdef void close_func(FT_Stream stream):
+cdef void close_func(FT_Stream stream) noexcept:
     """
     Close the stream.
 
--- a/renpy/text/hbfont.pyx
+++ b/renpy/text/hbfont.pyx
@@ -246,7 +246,7 @@ cdef bint is_zerowidth(unsigned int char):
 
     return False
 
-cdef unsigned long io_func(FT_Stream stream, unsigned long offset, unsigned char *buffer, unsigned long count):
+cdef unsigned long io_func(FT_Stream stream, unsigned long offset, unsigned char *buffer, unsigned long count) noexcept:
     """
     Seeks to offset, and then reads count bytes from the stream into buffer.
     """
@@ -283,7 +283,7 @@ cdef unsigned long io_func(FT_Stream stream, unsigned long offset, unsigned char
 
     return count
 
-cdef void close_func(FT_Stream stream):
+cdef void close_func(FT_Stream stream) noexcept:
     """
     Close the stream.
 
-- 
2.46.0

