--- a/src/importexport/musicxml/internal/musicxml/musicxmlsupport.h
+++ b/src/importexport/musicxml/internal/musicxml/musicxmlsupport.h
@@ -26,6 +26,7 @@
 #include <QDomElement>
 #include <QAbstractMessageHandler>
 #include <QSourceLocation>
+#include <QXmlStreamReader>
 
 #include "engraving/types/fraction.h"
 #include "engraving/dom/mscore.h"
--- a/src/project/internal/exporttype.h
+++ b/src/project/internal/exporttype.h
@@ -22,6 +22,9 @@
 #ifndef MU_PROJECT_EXPORTTYPE_H
 #define MU_PROJECT_EXPORTTYPE_H
 
+#include <QList>
+#include <QString>
+#include <QVariantList>
 #include <unordered_set>
 
 #include "containers.h"
