diff -upr a/Makefile b/Makefile
--- a/Makefile	2021-11-15 00:00:52.000000000 -0000
+++ b/Makefile	2023-05-21 03:36:55.052858431 -0000
@@ -16,7 +16,7 @@ linux:
 	$(CC) -shared -o libzest.so `find mruby/build/host -type f | grep -v mrbc | grep -e "\.o$$" | grep -v bin` ./deps/libnanovg.a \
 		./deps/libnanovg.a \
 		src/osc-bridge/libosc-bridge.a \
-		`pkg-config --libs libuv` -lm -lX11 -lGL -lpthread
+		`pkg-config --libs libuv` -lm -lX11 -lGL -lpthread -Wl,-soname,libzest.so
 	$(CC) test-libversion.c deps/pugl/pugl/pugl_x11.c \
 		 -DPUGL_HAVE_GL \
 		 -ldl -o zest -lX11 -lGL -lpthread -I deps/pugl -std=gnu99 -Wno-trigraphs

