--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -280,9 +280,15 @@ def rust_supported_targets(rustc):
     return data
 
 
+@imports('os')
+@imports(_from='mozbuild.util', _import='ensure_unicode')
+@imports(_from='mozbuild.util', _import='system_encoding')
 def detect_rustc_target(
     host_or_target, compiler_info, arm_target, rust_supported_targets
 ):
+    if "RUST_TARGET" in os.environ:
+        return ensure_unicode(os.environ["RUST_TARGET"], system_encoding)
+
     # Rust's --target options are similar to, but not exactly the same
     # as, the autoconf-derived targets we use.  An example would be that
     # Rust uses distinct target triples for targetting the GNU C++ ABI
