Only in b: carla-liblo-0.32.patch
diff -upr a/source/backend/CarlaPlugin.hpp b/source/backend/CarlaPlugin.hpp
--- a/source/backend/CarlaPlugin.hpp	2024-03-01 05:11:35.000000000 -0500
+++ b/source/backend/CarlaPlugin.hpp	2024-03-03 16:16:27.666090564 -0500
@@ -24,7 +24,11 @@
 // -----------------------------------------------------------------------
 // Avoid including extra libs here
 
+#ifdef HAVE_LIBLO
+#include <lo/lo_types.h>
+#else
 typedef void* lo_message;
+#endif
 typedef struct _NativePluginDescriptor NativePluginDescriptor;
 struct LADSPA_RDF_Descriptor;
 
