diff -upr a/source/Makefile.mk b/source/Makefile.mk
--- Carla-3b92085050c3dc00ac304e100ad648d55c067e65/source/Makefile.mk	2023-04-22 18:22:16.000000000 -0400
+++ Carla-3b92085050c3dc00ac304e100ad648d55c067e65-tmp/source/Makefile.mk	2023-04-23 20:33:10.654165789 -0400
@@ -84,12 +84,6 @@ BASE_FLAGS += -DNDEBUG $(BASE_OPTS) -fvi
 CXXFLAGS   += -fvisibility-inlines-hidden
 endif
 
-ifneq ($(MACOS_OR_WASM_OR_WINDOWS),true)
-ifneq ($(BSD),true)
-BASE_FLAGS += -fno-gnu-unique
-endif
-endif
-
 ifeq ($(WITH_LTO),true)
 BASE_FLAGS += -fno-strict-aliasing -flto
 LINK_OPTS  += -fno-strict-aliasing -flto -Werror=odr -Werror=lto-type-mismatch
