From 2fca3b88c565fffc6134faa67b75dfa4c70457b2 Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Wed, 12 Jun 2024 12:05:53 -0400
Subject: [PATCH] remove bpftool version check

--- a/meson.build
+++ b/meson.build
@@ -186,9 +186,6 @@ else
   bpftool_ver = run_command(get_bpftool_ver, bpftool_exe_path, check: true).stdout().strip()
   bpftool_maj = bpftool_ver.split('.')[0].to_int()
   bpftool_min = bpftool_ver.split('.')[1].to_int()
-  if bpftool_maj < 7 or (bpftool_maj == 7 and bpftool_min < 4)
-    error('bpftool >= 7.4 required (@0@ ver=@1@)'.format(bpftool_exe_path, bpftool_ver))
-  endif
   # this is a noop when we're not building bpftool ourselves
   bpftool_target = custom_target('bpftool_target',
               output: '@PLAINNAME@.__PHONY__',
-- 
2.45.2

