From 4bdf98cb5d456baa8f32c90e2a355fd2f6aac95c Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Sat, 4 May 2024 11:57:28 -0400
Subject: [PATCH] don't automatically strip during link

--- a/meson.build
+++ b/meson.build
@@ -313,7 +313,7 @@ else
         '-Wl,--as-needed,--no-undefined',
     ]
     link_flags_release += [
-        '-Wl,-O1,--gc-sections,--strip-all',
+        '-Wl,-O1,--gc-sections',
     ]
 endif
 
-- 
2.45.0

