--- a/libs/juce5/source/modules/juce_graphics/native/juce_freetype_Fonts.cpp
+++ b/libs/juce5/source/modules/juce_graphics/native/juce_freetype_Fonts.cpp
@@ -353,8 +353,8 @@ private:
     bool getGlyphShape (Path& destShape, const FT_Outline& outline, const float scaleX)
     {
         const float scaleY = -scaleX;
-        const short* const contours = outline.contours;
-        const char* const tags = outline.tags;
+        const unsigned short* const contours = outline.contours;
+        const unsigned char* const tags = outline.tags;
         const FT_Vector* const points = outline.points;
 
         for (int c = 0; c < outline.n_contours; ++c)
