From c327072b1cf9120b7f52b090f547d731aa401862 Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Tue, 6 Jun 2023 03:11:02 +0000
Subject: [PATCH] Fix build with clang (-fno-gnu-unique)

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index ecc6f501..d5940d52 100644
--- a/meson.build
+++ b/meson.build
@@ -144,7 +144,7 @@ if not os_darwin
     ]
 endif
 
-if not (os_darwin or os_windows)
+if cc.get_id() == 'gcc'
     build_flags += [
         '-fno-gnu-unique'
     ]
-- 
2.41.0

